tests/resolution/evaluation/test_evaluate_os_version.py

```diff
@@ -0,0 +1,79 @@
+"""Test OS Version evaluation."""
+
+from unittest.mock import PropertyMock, patch
+
+from awesomeversion import AwesomeVersion
+import pytest
+
+from supervisor.const import CoreState
+from supervisor.coresys import CoreSys
+from supervisor.os.manager import OSManager
+from supervisor.resolution.evaluations.os_version import EvaluateOSVersion
+
+
+@pytest.mark.parametrize(
+    "current,latest,expected",
+    [
+        ("10.0", "15.0", True),  # 5 major behind, should be unsupported
+        ("10.0", "14.0", False),  # 4 major behind, should be supported
+        ("10.2", "11.0", False),  # 1 major behind, supported
+        ("10.4", "10.5", False),  # same major, supported
+        ("10.5", "10.5", False),  # up to date, supported
+        ("10.5", "10.6", False),  # same major, supported
+        ("10.0", "13.3", False),  # 3 major behind, supported
+        (None, "15.0", False),  # No current version info, check skipped
+        ("2.0", None, False),  # No latest version info, check skipped
+        (
+            "9ccda431973acf17e4221850b08f3280b723df8d",
+            "15.0",
+            False,
+        ),  # Dev setup running on a commit hash, check skipped
```

Comment:
Maybe I'd add cases for some valid pre-release version strings in the mix, like `12.2.dev20240321` or `10.0.rc2`.

Reply:
Other than that, there are also branch dev builds which have Unix timestamp suffix instead of YYYYMMDD, e.g. `dev1744721960`.

Reply:
Ok I can drop this test then. I was just trying to think of something that might occur that tested the version exception. Figured building from a commit or branch was something done once in a while but maybe not.

IF there's a valid test which tests that exception handling let me know. IF not I can drop it. I can also drop the exception handling if it should never come up. I'd probably prefer not to log or capture because of the recurring point I mentioned above but if that's what you both want I'll add it. Realistically that is more about OS then Supervisor and what we'd expect in the wild for OS.

Reply:
> Figured building from a commit or branch was something done once in a while but maybe not.

That does happen, but builds from commit/branches still do not end up with version numbers which include commit hashes, at least the way the OS build scripts work currently. 

If we keep the exception handling discussed above, I agree then we should also keep that test case. So removing the git hash from this list should really be dependent on how the discussion above concludes.